dm/syncer/optimist.go

```diff
@@ -227,6 +248,39 @@ func (s *Syncer) handleQueryEventOptimistic(qec *queryEventContext) error {
 		}
 	}
 
+	s.resolveOptimisticDDL(qec.eventContext, getDDLJobSourceTable(job), job.targetTable, op.ConflictStage)
+
 	s.tctx.L().Info("finish to handle ddls in optimistic shard mode", zap.String("event", "query"), zap.Stringer("queryEventContext", qec))
 	return nil
 }
+
+func (s *Syncer) resolveOptimisticDDL(ec *eventContext, sourceTable, targetTable *filter.Table, stage optimism.ConflictStage) {
+	if sourceTable != nil && targetTable != nil {
+		if (stage == optimism.ConflictNone && s.osgk.tableInConflict(targetTable)) ||
+			s.osgk.inConflictStage(sourceTable, targetTable) {
+			// in the following two situations we should resolve this ddl lock at now
+			// 1. after this worker's ddl, the ddl lock is resolved
+			// 2. other worker has resolved this ddl lock, receives resolve command from master
+			group, redirectLocation := s.osgk.resolveGroup(targetTable)
+			if len(group) > 0 {
+				s.optimist.DoneRedirectOperation(utils.GenTableID(targetTable))
+				for tableID, location := range group {
+					s.saveTablePoint(utils.UnpackTableID(tableID), location)
```

Comment:
Is this line needed? Seems we have already saveTablePoint before handle that DDL.